arch/Kconfig

```
@@ -573,6 +573,13 @@ config IRQ_OFFLOAD
 	  run in interrupt context. Only useful for test cases that need
 	  to validate the correctness of kernel objects in IRQ context.
 
+config SRAM_VECTOR_TABLE
+	bool "Place the vector table in SRAM instead of flash"
+	depends on ARCH_HAS_VECTOR_TABLE_RELOCATION
```

Either here or at Cortex-M level:
```suggestion
	depends on ARCH_HAS_VECTOR_TABLE_RELOCATION || CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP
```
\+ see comment/proposal in `prep_c.c`

---

More on cortex-M level IMO like this:
```
select ARCH_HAS_VECTOR_TABLE_RELOCATION if (CPU_CORTEX_M_HAS_VTOR || CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP)
```
But since, as you said, SRAM_VECTOR_TABLE symbol will be set to y as you redefined the symbol dependencies, it will not add anything (or maybe transparency for a user).

This change might go with the changes in prep.c in the cae we don't have a VTOR register.

---

> More on cortex-M level IMO like this:
> 
> ```
> select ARCH_HAS_VECTOR_TABLE_RELOCATION if (CPU_CORTEX_M_HAS_VTOR || CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP)
> ```

LGTM.
 
> But since, as you said, SRAM_VECTOR_TABLE symbol will be set to y as you redefined the symbol dependencies, it will not add anything (or maybe transparency for a user).

One reason I was proposing this is that I'd like the SoC to *not* need to redefine the symbol dependencies :slightly_smiling_face: